### ISSUE TITLE
Indentation does not appear to be working.

### ISSUE THREAD

**Issue by chopfitzroy** [phalkunz/vim-ss#1]
I Am running this with neovim and when I try to indent all HTML is hard up on left side.

Very similar to this issue: http://stackoverflow.com/questions/459478/correct-indentation-of-html-and-php-using-vim

Any thoughts?

Cheers,
Otis.

**Comment by geekdenz** [comment]
I'm using 
https://github.com/othree/html5.vim.git
and this fixed it for me:
Change 
`vim-ss/indent/ss.vim`
to

```
setlocal indentexpr=SSIndent()

function! SSIndent()
    let line = getline(v:lnum)
    let previousNum = prevnonblank(v:lnum - 1)
    let previous = getline(previousNum)

    if (previous =~ "<%\\s*if.*%>" && previous !~ "<%\\s*end_if\\s*%>" && line !~ "<%\\s*end_if\\s*%>") 
        \ || (previous =~ "<%\\s*else.*%>" && previous !~ "<%\\s*end_else\\s*%>" && line !~ "    <%\\s*end_else\\s*%>") 
        \ || (previous =~ "<%\\s*else_if.*%>" && previous !~ "<%\\s*end_loop\\s*%>" && line !~ "<%\\s*end_loop\\s*%>") 
        \ || (previous =~ "<%\\s*loop.*%>" && previous !~ "<%\\s*end_loop\\s*%>" && line !~ "<%\\s*end_loop\\s*%>") 
        \ || (previous =~ "<%\\s*control.*%>" && previous !~ "<%\\s*end_control\\s*%>" && line !~ "<%\\s*end_control\\s*%>") 
        \ || (previous =~ "<%\\s*with.*%>" && previous !~ "<%\\s*end_with\\s*%>" && line !~ "<%\\s*end_with\\s*%>") 
        return indent(previousNum) + &tabstop
    else
        return HtmlIndentGet(v:lnum) " This is the change!
        "return HtmlIndent()
    endif
endfunction
```

Not sure how to fix it properly at the moment, but maybe it could be queried if HtmlIndentGet() is defined or something.

**Comment by chopfitzroy** [comment]
Hey @geekdenz sorry vim skills are a little rusty do you have a gist or dotfiles repo I can look at to see all of this in context?

Cheers.

**Comment by geekdenz** [comment]
Please see https://github.com/geekdenz/.dotfiles
and
https://github.com/geekdenz/vim-ss 

I fixed the issue there. But note that I think this requires the plugin
https://github.com/othree/html5.vim.git as far as I can tell.

**Comment by chopfitzroy** [comment]
Hey @geekdenz,

That's cool I use [vim-polygot](https://github.com/sheerun/vim-polyglot) which includes the html5 library anyways so that should not be an issue.

Cheers!

**Comment by chopfitzroy** [comment]
Hey @geekdenz,

Have you run into the issue where it is not indenting correctly on if statements i.e: `<% if $Query %>`. 

All the other tags (would you call them tags?) are working fine, just not these.

Cheers.

**Comment by geekdenz** [comment]
Yes, I've run into that issue as well. Will see if I can do something about that soon.

**Comment by chopfitzroy** [comment]
Hey @geekdenz, sweet let me know if I can help I am just pretty hopeless at `vim` as a language.

Cheers.

**Comment by phalkunz** [comment]
@geekdenz @CrashyBang Sorry guys, I haven't used this plugin for a while myself and I don't have time to look into this at the moment. Post  a solution when you have one or a pull request :) Thanks

**Comment by geekdenz** [comment]
Should I do a pull request? Haven't yet, because control statements are not indented. However, that is actually exactly what I wanted.

Should control statements be indented even?

**Comment by chopfitzroy** [comment]
Hey @geekdenz I don't really have an issue with no indentation one problem I did notice though was when you have an if wrapping a conditional html element the indenting breaks a little and looks like this:

```
<% if $First || $Odd %>
<div class="open">
    <% end_if %>
   <!-- inner code -->
    <% if $Last || $Even %>
</div>
<% end_if %>
```

I mean this is a pretty minor use case but it would be nice it we could fix to be more like the following (assuming no indentation):

```
<% if $First || $Odd %>
<div class="open">
<% end_if %>
   <!-- inner code -->
<% if $Last || $Even %>
</div>
<% end_if %>
```

**Comment by geekdenz** [comment]
Will have a look if/when I can.

**Comment by geekdenz** [comment]
Mh, I've played around a bit with searchpair() (see :h searchpair()), but tbh I'm not yet a master of vimscript and probably a long way away from being one. I guess, no pain, no gain, but this seems quite a hard problem to solve for the gain.

See https://github.com/geekdenz/vim-ss/blob/indent/indent/ss.vim .

This is because the HTML indenter is called otherwise. Maybe it can only be called when a block is only HTML with its closing elements. Might have to think a bit. 

Any hints welcome!
